### DISCUSSION TITLE
useProvider hook with biconomy

### DISCUSSION THREAD

**Discussion by dawksh** [wevm/wagmi#587] [Q&A]
I am trying to initialize biconomy mexa which takes a provider connected to the wallet, but the useProvider hook isn't returning a valid provider which does this, any idea how do I proceed?

**Comment by sjshon** [comment]
yeah i'm having this [problem ](https://github.com/wagmi-dev/wagmi/discussions/873) too. Any workaround?

**Reply by dawksh** [reply]
yes a pretty simple fix actually. if you're below mexa v2.0.33 then you can do 
```
const {data: signer} = useSigner()
const biconomy = new Biconomy(signer?.provider, {apiKey: apiKey});
```

and if you're on v3.0.0 or newer then you can do it like this: 
```
const {data: signer} = useSigner()

useEffect(() => {
 biconomy = new Biconomy((signer?.provider as any).provider, {apiKey: apiKey, contractAddresses: [<contract address>]})
})
```

**Reply by sjshon** [reply]
sounds great! would this be possible using `signer`?
`const web3 = new Web3(signer?.provider);`

**Reply by dawksh** [reply]
probably, you can use try `useProvider()` hook to get provider without any signer

**Reply by sjshon** [reply]
thanks @dawksh

**Reply by mohammedelgadi** [reply]
this is not working for me!
